src/BSON/Binary.c

```diff
@@ -142,6 +152,32 @@ static void php_phongo_binary_free_object(void *object TSRMLS_DC) /* {{{ */
 	efree(intern);
 } /* }}} */
 
+zend_object* php_phongo_binary_create_object(zend_class_entry *class_type TSRMLS_DC) /* {{{ */
+{
+        php_phongo_binary_t *intern;
+
+        intern = (php_phongo_binary_t *)ecalloc(1, sizeof(php_phongo_binary_t)+zend_object_properties_size(class_type));
+
+        zend_object_std_init(&intern->std, class_type TSRMLS_CC);
+        object_properties_init(&intern->std, class_type);
+
+        intern->std.handlers = &php_phongo_handler_binary;
+
+        return &intern->std;
+} /* }}} */
+#else
+static void php_phongo_binary_free_object(void *object TSRMLS_DC) /* {{{ */
+{
```

Comment:
That macro only works if the pointer is a zval containing an zend_object that is a php_phongo_binary_t* in this case we have the raw zend_object... So I think we need a new macro.

Compile errors with that macro:
In file included from /usr/include/php/Zend/zend.h:31:0,
                 from /usr/include/php/main/php.h:35,
                 from /root/mongo-php-driver-prototype/src/BSON/Binary.c:32:
/root/mongo-php-driver-prototype/src/BSON/Binary.c: In function 'php_phongo_binary_free_object':
/usr/include/php/Zend/zend_types.h:529:31: error: 'zend_object' has no member named 'value'
 #define Z_OBJ(zval)     (zval).value.obj
                               ^
/usr/include/php/Zend/zend_types.h:530:28: note: in expansion of macro 'Z_OBJ'
 #define Z_OBJ_P(zval_p)    Z_OBJ(*(zval_p))
                            ^
./php_phongo_classes.h:36:64: note: in expansion of macro 'Z_OBJ_P'
 # define Z_BINARY_OBJ_P(zv)            php_binary_fetch_object(Z_OBJ_P(zv));
                                                                ^
/root/mongo-php-driver-prototype/src/BSON/Binary.c:139:32: note: in expansion of macro 'Z_BINARY_OBJ_P'
  php_phongo_binary_t *intern = Z_BINARY_OBJ_P(object);
                                ^
make: *** [src/BSON/Binary.lo] Error 1